projects/components/src/number-input/number-input.component.ts

```diff
@@ -67,7 +67,7 @@ export class NumberInputComponent implements ControlValueAccessor {
     this.value = value;
 
     this.enforceMinMaxAndEmit();
-    this.propagateValueChangeToFormControl(value);
+    this.propagateValueChangeToFormControl(this.value);
```

Comment:
what difference does it make?

Reply:
local value could be update it in 'EnforceMinMaxAndEmit' for this reason is not fine to propagate the value without this specific change.

Reply:
Hmm.. might want to rearrange the code here to make it more obvious. Also, please put a comment here explaining this requirement

Reply:
@anandtiwary I refactored the component to keep the logic clearer. Please take a look and let me know if you agree with this.